src/ZoneServer/Buffs/Base/BuffHandler.cs

```diff
@@ -25,6 +25,14 @@ public virtual void OnStart(Buff buff)
 		{
 		}
 
+		/// <summary>
+		/// Called when overbuffing
+		/// </summary>
+		/// <param name="buff"></param>
+		public virtual void OnOverbuff(Buff buff)
```

Comment:
Is this what happens?  I was looking at BuffComponent.Overbuff, which doesn't call Start if the overbuff counter doesn't change.  I see that BuffComponent.Add always calls start when overbuffing, which is confusing.  Are both of these called at different times?

Reply:
`BuffComponent.Add` always calls `Start` because it replaces the buff if it's already there. The purpose is different, because it treats the buff as a brand new buff. Though you're right that `Start` won't be called if the overbuff counter didn't actually increase. Do you need that? A callback for when a buff is "applied"? Regardless of whether it's started, overbuffed, or just extended?

Reply:
`BuffComponent.Add` always calls `Start` because it has a different purpose. It treats the buff as a brand new buff and discards any existing buffs with the same id. You're right that `Start` is not called if the overbuff counter doesn't increase though. Do you need that? A callback for when a buff is "applied", regardless of whether it's started, overbuffed, or just extended?

I was actually just debating this with someone else, who had a potential use case for such a callback as well, but I wasn't sure whether we actually need that.

Reply:
I tested to see if I could rewrite it not to use OnOverbuff, and it is possible, so I suppose that feature is not currently needed.